src/run/formatters.ts

```diff
@@ -56,8 +60,8 @@ export async function initializeFormatters({
       ).isTTY
     }
     if (type === 'progress-bar' && !(stream as TtyWriteStream).isTTY) {
-      console.warn(
-        `Cannot use 'progress-bar' formatter for output to '${target}' as not a TTY. Switching to 'progress' formatter.`
+      stderr.write(
```

Comment:
Where we currently `warn` it would be good to maintain that visually.

We could consider creating a `new Console` with the stdout and stderr streams and using that?

Reply:
it's definitely better to be connected to an injected `stderr` instance than the global `console`, but I agree the semantics aren't as nice.

It's tempting to suggest a `UserInterface` abstraction, but that's maybe pushing it a bit far since these formatters are only built to work on a console-based UI. I like the idea of an intermediate `Console` class though. The `warn` method could also colour the output, for example.

Reply:
@mattwynne this [already exists in Node](https://nodejs.org/api/console.html#class-console), so we can do `new Console(stdout, stderr)`.

Reply:
Discovered `warn` is just an alias for `error` so it doesn't do anything differently vs error. 

I'm still okay passing a console instead of stderr. Will make that update

Reply:
Feels like if we pass in `stdout` and `stderr` to `runCucumber` but then do `const logger = new Console(stdout, stderr)` and pass that along to lower lever stuff that would be a good balance.